libopenage/engine.cpp

```diff
@@ -548,4 +542,12 @@ void Engine::move_phys_camera(float x, float y, float amount) {
 	this->engine_coord_data->camgame_phys += cam_delta.to_phys3();
 }
 
+void Engine::start_game(std::unique_ptr<GameMain> game) {
+	// TODO: maybe implement a proper 1-to-1 connection
+	assert(game);
```

Comment:
Ok, we have that:

there are errors, caused by the bug and errors, caused by the user
the former should propagate to the main and stop the program
the latter should be handled and print error dialogs or whatever
and for both we can use our current exception system

What about the syntax? You're saying that it's "`if() throw Error()`"? Inability to use them liberally will discourage checking and documenting.